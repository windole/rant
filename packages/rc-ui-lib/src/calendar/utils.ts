/* eslint-disable no-nested-ternary */
export function compareMonth(date1: Date, date2: Date): number {
  const year1 = date1.getFullYear();
  const year2 = date2.getFullYear();

  if (year1 === year2) {
    const month1 = date1.getMonth();
    const month2 = date2.getMonth();
    return month1 === month2 ? 0 : month1 > month2 ? 1 : -1;
  }

  return year1 > year2 ? 1 : -1;
}

export function compareDay(day1: Date, day2: Date): number {
  const compareMonthResult = compareMonth(day1, day2);

  if (compareMonthResult === 0) {
    const date1 = day1.getDate();
    const date2 = day2.getDate();
    return date1 === date2 ? 0 : date1 > date2 ? 1 : -1;
  }

  return compareMonthResult;
}

export const cloneDate = (date: Date): Date => new Date(date);

export const cloneDates = (dates: Date | Date[]): Date | Date[] =>
  Array.isArray(dates) ? dates.map(cloneDate) : cloneDate(dates);

export function getDayByOffset(date: Date, offset: number): Date {
  const cloned = cloneDate(date);
  cloned.setDate(cloned.getDate() + offset);
  return cloned;
}

export const getPrevDay = (date: Date): Date => getDayByOffset(date, -1);
export const getNextDay = (date: Date): Date => getDayByOffset(date, 1);
export const getToday = (): Date => {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  return today;
};

export function calcDateNum(date: [Date, Date]): number {
  const day1 = date[0].getTime();
  const day2 = date[1].getTime();
  return (day2 - day1) / (1000 * 60 * 60 * 24) + 1;
}

export const getMonthEndDay = (year: number, month: number): number =>
  32 - new Date(year, month - 1, 32).getDate();

const calendarLang = {
  end: '结束',
  start: '开始',
  title: '日期选择',
  confirm: '确定',
  startEnd: '开始/结束',
  weekdays: ['日', '一', '二', '三', '四', '五', '六'],
  monthTitle: (year: number, month: number) => `${year}年${month}月`,
  rangePrompt: (maxRange: number) => `最多选择 ${maxRange} 天`,
};

export const t = (key: string, ...args): string => {
  if (args.length) {
    return calendarLang[key](...args);
  }
  return calendarLang[key];
};

export const formatMonthTitle = (date: Date): string =>
  t('monthTitle', date.getFullYear(), date.getMonth() + 1);
