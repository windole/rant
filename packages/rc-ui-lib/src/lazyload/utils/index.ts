import { ListenEvent } from '../PropsType';

/* eslint-disable prefer-rest-params */
const inBrowser = typeof window !== 'undefined';

export const DEFAULT_URL =
  'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';

// 需要监听的事件
export const DEFAULT_EVENTS: ListenEvent[] = [
  'scroll',
  'wheel',
  'resize',
  'mousewheel',
  'animationend',
  'transitionend',
  'touchmove',
];

export const hasIntersectionObserver =
  inBrowser &&
  'IntersectionObserver' in window &&
  'IntersectionObserverEntry' in window &&
  'intersectionRatio' in window.IntersectionObserverEntry.prototype;

export const modeType = {
  event: 'event',
  observer: 'observer',
};

export const call = (fn): void => fn && fn();

export function setImage(
  element: HTMLMediaElement | HTMLIFrameElement | HTMLImageElement | HTMLElement,
  imagePath: string,
): void {
  if (
    element instanceof HTMLIFrameElement ||
    element instanceof HTMLImageElement ||
    element instanceof HTMLMediaElement
  ) {
    element.src = imagePath;
  } else {
    element.style.backgroundImage = `url('${imagePath}')`;
  }
}

export function loadImage(imagePath: string): Promise<string> {
  return new Promise((resolve, reject) => {
    const img = new Image();
    img.src = imagePath;
    img.onload = () => resolve(imagePath);
    img.onerror = (err) => reject(err);
  });
}

export function addCssClassName(
  element: HTMLImageElement | HTMLElement,
  cssClassName: string,
): void {
  if (!element.className.includes(cssClassName)) {
    element.className += ` ${cssClassName}`;
  }
}

export function throttle(fn, threshold = 250, scope?) {
  let last;
  let deferTimer;
  return () => {
    // @ts-ignore
    const context = scope || this;

    const now = +new Date();
    const args = arguments;
    if (last && now < last + threshold) {
      // hold on to it
      clearTimeout(deferTimer);
      deferTimer = setTimeout(() => {
        last = now;
        fn.apply(context, args);
      }, threshold);
    } else {
      last = now;
      fn.apply(context, args);
    }
  };
}

export function debounce(func, wait, immediate?) {
  let timeout;
  let args;
  let context;
  let timestamp;
  let result;

  const later = function later() {
    const last = +new Date() - timestamp;

    if (last < wait && last >= 0) {
      timeout = setTimeout(later, wait - last);
    } else {
      timeout = null;
      if (!immediate) {
        result = func.apply(context, args);
        if (!timeout) {
          context = null;
          args = null;
        }
      }
    }
  };

  return function debounced() {
    // @ts-ignore
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    context = this;
    args = arguments;
    timestamp = +new Date();

    const callNow = immediate && !timeout;
    if (!timeout) {
      timeout = setTimeout(later, wait);
    }

    if (callNow) {
      result = func.apply(context, args);
      context = null;
      args = null;
    }

    return result;
  };
}
