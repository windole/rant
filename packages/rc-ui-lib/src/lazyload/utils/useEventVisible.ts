import { useEffect, useRef, useState } from 'react';
import { debounce as debounceFunc, DEFAULT_EVENTS, throttle as throttleFunc } from '.';
import { EventOptions } from '../PropsType';

// try to handle passive events
let passiveEventSupported = false;
try {
  const opts = Object.defineProperty({}, 'passive', {
    get() {
      passiveEventSupported = true;
    },
  });
  window.addEventListener('test', null, opts);
  // eslint-disable-next-line no-empty
} catch (e) {}

const passiveEvent = passiveEventSupported ? { capture: false, passive: true } : false;

const useEventVisible = (
  eventOptions: EventOptions,
): [React.MutableRefObject<HTMLDivElement | HTMLImageElement>, boolean] => {
  const {
    listenEvents = DEFAULT_EVENTS,
    offset,
    scrollContainer,
    debounce,
    throttle,
  } = eventOptions;
  // 是否可见
  const [visible, setVisible] = useState(false);
  // // 包裹容器
  const containerRef = useRef<HTMLDivElement | HTMLImageElement>(null);

  useEffect(() => {
    // 检查函数
    let checkVisible = () => {
      // 如果容器不存在则不计算
      if (!containerRef.current) return;
      // 获取当前组件位置
      const { top, left, width, height } = containerRef.current?.getBoundingClientRect();
      // 获取滚动容器位置
      const {
        top: parentTop,
        left: parentLeft,
        width: parentWidth,
        height: parentHeight,
      } = scrollContainer.getBoundingClientRect();
      // 计算屏幕高度和宽度
      const windowInnerHeight = window.innerHeight || document.documentElement.clientHeight;
      const windowInnerWidth = window.innerWidth || document.documentElement.clientWidth;
      // 计算滚动容器和屏幕可视区的相交区域
      const intersectionTop = Math.min(parentTop, 0);
      const intersectionLeft = Math.min(parentLeft, 0);

      const intersectionHeight =
        Math.min(windowInnerHeight, parentTop + parentHeight) - intersectionTop;
      const intersectionWidth =
        Math.min(windowInnerWidth, parentLeft + parentWidth) - intersectionLeft;
      // 计算组件距离可视区的高度
      const offsetTop = top - intersectionTop;
      const offsetLeft = left - intersectionLeft;
      // 偏移量计算[top, left]
      const offsets = Array.isArray(offset) ? offset : [offset, offset];
      // 通过上述距离判断组件是否在可见区域
      const isVisible =
        offsetTop - offsets[0] <= intersectionHeight &&
        offsetTop + height + offsets[0] >= 0 &&
        offsetLeft - offsets[1] <= intersectionWidth &&
        offsetLeft + width + offsets[1] >= 0;

      setVisible(isVisible);
      // 如果可见取消当前组件的所有的监听
      if (isVisible) {
        listenEvents.forEach((event) => {
          scrollContainer?.removeEventListener(event, checkVisible);
        });
      }
    };

    // 使用防抖节流增强checkVisible函数，这里优先防抖，其次节流
    if (debounce) {
      // 防抖函数
      checkVisible = debounceFunc(checkVisible, debounce);
    } else if (throttle) {
      // 节流函数
      checkVisible = throttleFunc(checkVisible, throttle);
    }
    // 批量监听事件
    listenEvents.forEach((event) => {
      scrollContainer?.addEventListener(event, checkVisible, passiveEvent);
    });

    // 自动执行一次
    checkVisible();

    return () => {
      listenEvents.forEach((event) => {
        scrollContainer?.removeEventListener(event, checkVisible, passiveEvent);
      });
    };
  }, [scrollContainer]);
  return [containerRef, visible];
};

export default useEventVisible;
