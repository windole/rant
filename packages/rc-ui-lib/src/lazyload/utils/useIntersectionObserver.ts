import { useCallback, useEffect, useState } from 'react';

const useIntersectionObserver = (
  observerOptions: IntersectionObserverInit,
  handleIntersection: IntersectionObserverCallback,
): ((node: Element) => void) => {
  const [element, setElement] = useState<Element>(null);

  const handleElement = useCallback((node: Element) => {
    setElement(node);
  }, []);

  useEffect(() => {
    if (!element) return;
    const observer = new IntersectionObserver(handleIntersection, observerOptions);
    observer.observe(element);
    // eslint-disable-next-line consistent-return
    return () => observer && observer.disconnect && observer.disconnect();
  }, [handleIntersection, observerOptions, element]);
  return handleElement;
};

export default useIntersectionObserver;
