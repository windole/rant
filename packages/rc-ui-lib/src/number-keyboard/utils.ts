import { KeyConfig, NumberKeyboardProps } from './PropsType';

const genBasicKeys = (props: NumberKeyboardProps) => {
  const keys: KeyConfig[] = Array(9)
    .fill('')
    .map((_, i) => ({ text: i + 1 }));

  if (props.randomKeyOrder) {
    keys.sort(() => (Math.random() > 0.5 ? 1 : -1));
  }

  return keys;
};

export const genDefaultKeys = (props: NumberKeyboardProps): KeyConfig[] => [
  ...genBasicKeys(props),
  { text: props.extraKey as string, type: 'extra' },
  { text: 0 },
  {
    text: props.showDeleteKey ? props.deleteButtonText : '',
    type: props.showDeleteKey ? 'delete' : '',
  },
];

export const genCustomKeys = (props: NumberKeyboardProps) => {
  const keys = genBasicKeys(props);
  const { extraKey } = props;
  const extraKeys = Array.isArray(extraKey) ? extraKey : [extraKey];

  if (extraKeys.length === 1) {
    keys.push({ text: 0, wider: true }, { text: extraKeys[0], type: 'extra' });
  } else if (extraKeys.length === 2) {
    keys.push(
      { text: extraKeys[0], type: 'extra' },
      { text: 0 },
      { text: extraKeys[1], type: 'extra' },
    );
  }

  return keys;
};
