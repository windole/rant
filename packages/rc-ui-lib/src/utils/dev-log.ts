export function devWarning(component: string, message: string): void {
  if (process.env.NODE_ENV !== 'production') {
    console.warn(`[${component}] ${message}`);
  }
}

export function devError(component: string, message: string) {
  if (process.env.NODE_ENV !== 'production') {
    console.error(`[${component}] ${message}`);
  }
}
