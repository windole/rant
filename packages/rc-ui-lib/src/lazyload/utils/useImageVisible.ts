import { useCallback, useState } from 'react';
import { addCssClassName, loadImage, setImage } from '.';
import { IntersectionObserverEntryType, ObserverOptions } from '../PropsType';
import useIntersectionObserver from './useIntersectionObserver';

const defaultObserverOptions = {
  threshold: [0, 1],
};

const useImageVisible = (
  observerOptions: ObserverOptions = defaultObserverOptions,
  errorImage: string,
): [(node: Element) => void, boolean] => {
  const [isLoaded, setIsLoaded] = useState(false);

  const handleIntersection = useCallback(
    (entries: IntersectionObserverEntryType[]) => {
      entries
        .filter((entry) => entry.isIntersecting)
        .forEach((entry) => {
          const target = entry.target as HTMLImageElement | HTMLElement;
          const url = target.dataset.src || '';
          loadImage(url)
            .catch(() => {
              setImage(target, errorImage);
              addCssClassName(target, 'rc-lazyload__error');
              target.setAttribute('lazy', 'error');
            })
            .then((imagePath: string) => {
              setImage(target, imagePath);
              setIsLoaded(true);
              addCssClassName(target, 'rc-lazyload__loaded');
              target.setAttribute('lazy', 'loaded');
              target.removeAttribute('data-src');
            });
        });
    },
    [setIsLoaded, isLoaded],
  );
  const ref = useIntersectionObserver(observerOptions, handleIntersection);
  return [ref, isLoaded];
};

export default useImageVisible;
