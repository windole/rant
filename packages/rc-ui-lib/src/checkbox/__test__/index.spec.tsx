import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Checkbox } from '..';

import mountTest from '../../../tests/shared/mountTest';

import { sleep } from '../../../tests/utils';
import {
  CheckboxGroupInstance,
  CheckboxGroupToggleAllOptions,
  CheckboxInstance,
} from '../PropsType';

describe('Checkbox', () => {
  mountTest(Checkbox);
  mountTest(() => (
    <Checkbox checked onChange={jest.fn()}>
      Label
    </Checkbox>
  ));

  it('renders correctly', () => {
    expect(
      <Checkbox defaultChecked onChange={jest.fn()}>
        Label
      </Checkbox>,
    ).toMatchRenderedSnapshot();
  });

  it('render Custom icon correctly', () => {
    const activeIcon = 'https://img.yzcdn.cn/vant/user-active.png';
    const inactiveIcon = 'https://img.yzcdn.cn/vant/user-inactive.png';
    expect(
      <Checkbox
        defaultChecked
        iconRender={({ checked: isActive }) => (
          <img alt="" src={isActive ? activeIcon : inactiveIcon} />
        )}
      >
        自定义图标
      </Checkbox>,
    ).toMatchRenderedSnapshot();
  });

  it('should emit change event when checkbox icon is clicked', async () => {
    const onChange = jest.fn();
    const wrapper = mount(<Checkbox onChange={onChange}>Label</Checkbox>);

    const icon = wrapper.find('.rc-checkbox__icon');
    icon.simulate('click');
    expect(onChange).toBeCalledWith(true);

    await wrapper.setProps({ checked: true });
    icon.simulate('click');
    expect(onChange).toBeCalledWith(false);
  });

  it('should emit toggle event when checkbox is clicked', async () => {
    // const onToggle = jest.fn();
    const checkboxRef = React.createRef<CheckboxInstance>();
    const toggle = (checked?: boolean) => {
      checkboxRef.current.toggle(checked);
    };
    const wrapper = mount(<Checkbox ref={checkboxRef}>Label</Checkbox>);

    await act(async () => {
      toggle();
    });
    wrapper.update();
    await sleep();
    expect(wrapper.childAt(0).props().checked).toEqual(true);
    // expect(onToggle).toBeCalled();
  });

  it('should not emit "change" event when checkbox icon is disabled and clicked', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Checkbox defaultChecked disabled onChange={onChange}>
        Label
      </Checkbox>,
    );

    wrapper.find('.rc-checkbox__icon').simulate('click');
    expect(onChange).not.toHaveBeenCalled();
  });

  it('should render "rc-checkbox--label-disabled" class when using label-disabled prop', () => {
    const wrapper = mount(
      <Checkbox defaultChecked labelDisabled onChange={jest.fn()}>
        Label
      </Checkbox>,
    );

    expect(wrapper.find('.rc-checkbox--label-disabled')).toBeTruthy();
  });

  it('should emit "change" event when label is clicked', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Checkbox defaultChecked onChange={onChange}>
        Label
      </Checkbox>,
    );

    const icon = wrapper.find('.rc-checkbox__label');
    icon.simulate('click');
    expect(onChange).toBeCalledWith(false);
  });

  it('should not emit "change" event when label is disabled and clicked', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Checkbox defaultChecked labelDisabled onChange={onChange}>
        Label
      </Checkbox>,
    );

    const icon = wrapper.find('.rc-checkbox__label');
    icon.simulate('click');
    expect(onChange).not.toHaveBeenCalled();
  });

  it('should adjust label position when using label-position prop', () => {
    const wrapper = mount(
      <Checkbox defaultChecked labelPosition="left" onChange={jest.fn()}>
        Label
      </Checkbox>,
    );

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should emit click event when checkbox icon is clicked', async () => {
    const onClick = jest.fn();
    const wrapper = mount(
      <Checkbox defaultChecked labelPosition="left" onClick={onClick}>
        Label
      </Checkbox>,
    );

    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);

    const icon = wrapper.find('.rc-checkbox__icon');
    icon.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(2);
  });
});

describe('Checkbox.Group', () => {
  mountTest(Checkbox.Group);

  it('renders correctly', () => {
    const wrapper = mount(
      <Checkbox.Group value={['1']}>
        <Checkbox name="0">选项一</Checkbox>
        <Checkbox name="1" checked>
          选项二
        </Checkbox>
        <Checkbox name="2">选项三</Checkbox>
      </Checkbox.Group>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render value correctly', () => {
    const wrapper = mount(
      <Checkbox.Group value={['0']}>
        <Checkbox name="0">选项一</Checkbox>
        <Checkbox name="1">选项二</Checkbox>
        <Checkbox name="2">选项三</Checkbox>
      </Checkbox.Group>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should change icon size when using icon-size prop', () => {
    const wrapper = mount(
      <Checkbox.Group value={['1']} iconSize="10rem">
        <Checkbox name="0">选项一</Checkbox>
        <Checkbox name="1" iconSize="5rem" checked>
          选项二
        </Checkbox>
      </Checkbox.Group>,
    );

    const icons = wrapper.find('.rc-checkbox__icon');
    // expect(icons).toHaveLength(2);
    // console.log(icons.debug());
    // console.log(icons.at(0).getDOMNode());
    // console.log(icons.at(0).getDOMNode());
    expect(icons.at(0).getDOMNode().style.fontSize).toEqual('10rem');
    expect(icons.at(1).getDOMNode().style.fontSize).toEqual('5rem');
  });

  it('should limit change num when using max prop', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <Checkbox.Group value={['a', 'b']} max={2} onChange={onChange}>
        <Checkbox name="a">选项一</Checkbox>
        <Checkbox name="b">选项二</Checkbox>
        <Checkbox name="c">选项三</Checkbox>
      </Checkbox.Group>,
    );

    const icons = wrapper.find('.rc-checkbox__icon');
    icons.at(2).simulate('click');
    expect(onChange).toHaveBeenCalledTimes(0);
    icons.at(0).simulate('click');
    expect(onChange).toHaveBeenCalledTimes(1);
    icons.at(1).simulate('click');
    expect(onChange).toHaveBeenCalledTimes(2);
  });

  it('should change checked color when using checked-color prop', () => {
    const wrapper = mount(
      <Checkbox.Group value={['a', 'b']} checkedColor="black">
        <Checkbox name="a">选项一</Checkbox>
        <Checkbox name="b" checkedColor="white">
          选项二
        </Checkbox>
      </Checkbox.Group>,
    );

    const icons = wrapper.find('i.van-icon');
    // console.log(icons.at(0).getDOMNode().style);
    // console.log(icons.at(1).getDOMNode().style);
    expect(icons).toHaveLength(2);
    expect(icons.at(0).getDOMNode().style.backgroundColor).toEqual('black');
    expect(icons.at(1).getDOMNode().style.backgroundColor).toEqual('white');
  });

  it('should ignore Checkbox if bind-group is false', async () => {
    const onChange = jest.fn();
    const groupValue = [];
    const groupRef = React.createRef<CheckboxGroupInstance>();
    const toggleAll = (checked?: boolean) => {
      groupRef.current.toggleAll(checked);
    };
    const wrapper = mount(
      <Checkbox.Group value={groupValue} ref={groupRef} onChange={onChange}>
        <Checkbox checked={false} name="a" bindGroup={false} />
        <Checkbox name="b" />
        <Checkbox name="c" />
      </Checkbox.Group>,
    );

    const items = wrapper.find('.rc-checkbox');
    items.at(0).simulate('click');
    expect(wrapper.props().value).toEqual([]);
    expect(onChange).not.toHaveBeenCalled();

    await act(async () => {
      toggleAll(true);
    });
    await sleep(0);
    expect(onChange).toBeCalledWith(['b', 'c']);
  });

  it('should toggle all checkboxes when toggleAll method is called', async () => {
    const onChange = jest.fn();
    const groupValue = [];
    const groupRef = React.createRef<CheckboxGroupInstance>();
    const toggleAll = (options?: CheckboxGroupToggleAllOptions) => {
      groupRef.current.toggleAll(options);
    };
    mount(
      <Checkbox.Group value={groupValue} ref={groupRef} onChange={onChange}>
        <Checkbox name="a" />
        <Checkbox name="b" />
        <Checkbox name="c" disabled />
      </Checkbox.Group>,
    );

    await act(async () => {
      toggleAll();
    });
    await sleep(0);
    expect(onChange).toBeCalledWith(['a', 'b', 'c']);

    await act(async () => {
      toggleAll(false);
    });
    await sleep(0);
    expect(onChange).toBeCalledWith([]);

    await act(async () => {
      toggleAll(true);
    });
    await sleep(0);
    expect(onChange).toBeCalledWith(['a', 'b', 'c']);

    await act(async () => {
      toggleAll({ checked: true, skipDisabled: true });
    });
    await sleep(0);
    expect(onChange).toBeCalledWith(['a', 'b']);
  });
});
