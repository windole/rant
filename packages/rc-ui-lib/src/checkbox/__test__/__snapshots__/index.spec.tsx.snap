// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render Custom icon correctly 1`] = `
<div
  aria-checked="true"
  class="rc-checkbox"
  role="checkbox"
  tabindex="0"
>
  <div
    class="rc-checkbox__icon rc-checkbox__icon--round rc-checkbox__icon--checked"
  >
    <img
      alt=""
      src="https://img.yzcdn.cn/vant/user-active.png"
    />
  </div>
  <span
    class="rc-checkbox__label"
  >
    自定义图标
  </span>
</div>
`;

exports[`Checkbox renders correctly 1`] = `
<div
  aria-checked="true"
  class="rc-checkbox"
  role="checkbox"
  tabindex="0"
>
  <div
    class="rc-checkbox__icon rc-checkbox__icon--round rc-checkbox__icon--checked"
  >
    <i
      class="rc-badge__wrapper van-icon van-icon-success"
    />
  </div>
  <span
    class="rc-checkbox__label"
  >
    Label
  </span>
</div>
`;

exports[`Checkbox should adjust label position when using label-position prop 1`] = `"<div role=\\"checkbox\\" class=\\"rc-checkbox\\" tabindex=\\"0\\" aria-checked=\\"true\\"><span class=\\"rc-checkbox__label rc-checkbox__label--left\\">Label</span><div class=\\"rc-checkbox__icon rc-checkbox__icon--round rc-checkbox__icon--checked\\"><i class=\\"rc-badge__wrapper van-icon van-icon-success\\"></i></div></div>"`;

exports[`Checkbox.Group render value correctly 1`] = `
<ForwardRef
  defaultValue={Array []}
  value={
    Array [
      "0",
    ]
  }
>
  <div
    className="rc-checkbox-group"
  >
    <Checkbox
      bindGroup={true}
      key=".0"
      name="0"
    >
      <Checker
        bem={[Function]}
        bindGroup={true}
        checked={true}
        name="0"
        onToggle={[Function]}
        parent={
          Object {
            "props": Object {
              "children": Array [
                <Checkbox
                  bindGroup={true}
                  name="0"
                >
                  选项一
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="1"
                >
                  选项二
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="2"
                >
                  选项三
                </Checkbox>,
              ],
              "defaultValue": Array [],
              "value": Array [
                "0",
              ],
            },
          }
        }
        role="checkbox"
        shape="round"
      >
        <div
          aria-checked={true}
          className="rc-checkbox"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
        >
          <div
            className="rc-checkbox__icon rc-checkbox__icon--round rc-checkbox__icon--checked"
            style={
              Object {
                "fontSize": undefined,
              }
            }
          >
            <Icon
              name="success"
              style={Object {}}
              tag="i"
            >
              <Badge
                className="van-icon van-icon-success"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-success"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
          </div>
          <span
            className="rc-checkbox__label"
          >
            选项一
          </span>
        </div>
      </Checker>
    </Checkbox>
    <Checkbox
      bindGroup={true}
      key=".1"
      name="1"
    >
      <Checker
        bem={[Function]}
        bindGroup={true}
        checked={false}
        name="1"
        onToggle={[Function]}
        parent={
          Object {
            "props": Object {
              "children": Array [
                <Checkbox
                  bindGroup={true}
                  name="0"
                >
                  选项一
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="1"
                >
                  选项二
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="2"
                >
                  选项三
                </Checkbox>,
              ],
              "defaultValue": Array [],
              "value": Array [
                "0",
              ],
            },
          }
        }
        role="checkbox"
        shape="round"
      >
        <div
          aria-checked={false}
          className="rc-checkbox"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
        >
          <div
            className="rc-checkbox__icon rc-checkbox__icon--round"
            style={
              Object {
                "fontSize": undefined,
              }
            }
          >
            <Icon
              name="success"
              style={Object {}}
              tag="i"
            >
              <Badge
                className="van-icon van-icon-success"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-success"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
          </div>
          <span
            className="rc-checkbox__label"
          >
            选项二
          </span>
        </div>
      </Checker>
    </Checkbox>
    <Checkbox
      bindGroup={true}
      key=".2"
      name="2"
    >
      <Checker
        bem={[Function]}
        bindGroup={true}
        checked={false}
        name="2"
        onToggle={[Function]}
        parent={
          Object {
            "props": Object {
              "children": Array [
                <Checkbox
                  bindGroup={true}
                  name="0"
                >
                  选项一
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="1"
                >
                  选项二
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="2"
                >
                  选项三
                </Checkbox>,
              ],
              "defaultValue": Array [],
              "value": Array [
                "0",
              ],
            },
          }
        }
        role="checkbox"
        shape="round"
      >
        <div
          aria-checked={false}
          className="rc-checkbox"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
        >
          <div
            className="rc-checkbox__icon rc-checkbox__icon--round"
            style={
              Object {
                "fontSize": undefined,
              }
            }
          >
            <Icon
              name="success"
              style={Object {}}
              tag="i"
            >
              <Badge
                className="van-icon van-icon-success"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-success"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
          </div>
          <span
            className="rc-checkbox__label"
          >
            选项三
          </span>
        </div>
      </Checker>
    </Checkbox>
  </div>
</ForwardRef>
`;

exports[`Checkbox.Group renders correctly 1`] = `
<ForwardRef
  defaultValue={Array []}
  value={
    Array [
      "1",
    ]
  }
>
  <div
    className="rc-checkbox-group"
  >
    <Checkbox
      bindGroup={true}
      key=".0"
      name="0"
    >
      <Checker
        bem={[Function]}
        bindGroup={true}
        checked={false}
        name="0"
        onToggle={[Function]}
        parent={
          Object {
            "props": Object {
              "children": Array [
                <Checkbox
                  bindGroup={true}
                  name="0"
                >
                  选项一
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  checked={true}
                  name="1"
                >
                  选项二
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="2"
                >
                  选项三
                </Checkbox>,
              ],
              "defaultValue": Array [],
              "value": Array [
                "1",
              ],
            },
          }
        }
        role="checkbox"
        shape="round"
      >
        <div
          aria-checked={false}
          className="rc-checkbox"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
        >
          <div
            className="rc-checkbox__icon rc-checkbox__icon--round"
            style={
              Object {
                "fontSize": undefined,
              }
            }
          >
            <Icon
              name="success"
              style={Object {}}
              tag="i"
            >
              <Badge
                className="van-icon van-icon-success"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-success"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
          </div>
          <span
            className="rc-checkbox__label"
          >
            选项一
          </span>
        </div>
      </Checker>
    </Checkbox>
    <Checkbox
      bindGroup={true}
      checked={true}
      key=".1"
      name="1"
    >
      <Checker
        bem={[Function]}
        bindGroup={true}
        checked={true}
        name="1"
        onToggle={[Function]}
        parent={
          Object {
            "props": Object {
              "children": Array [
                <Checkbox
                  bindGroup={true}
                  name="0"
                >
                  选项一
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  checked={true}
                  name="1"
                >
                  选项二
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="2"
                >
                  选项三
                </Checkbox>,
              ],
              "defaultValue": Array [],
              "value": Array [
                "1",
              ],
            },
          }
        }
        role="checkbox"
        shape="round"
      >
        <div
          aria-checked={true}
          className="rc-checkbox"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
        >
          <div
            className="rc-checkbox__icon rc-checkbox__icon--round rc-checkbox__icon--checked"
            style={
              Object {
                "fontSize": undefined,
              }
            }
          >
            <Icon
              name="success"
              style={Object {}}
              tag="i"
            >
              <Badge
                className="van-icon van-icon-success"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-success"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
          </div>
          <span
            className="rc-checkbox__label"
          >
            选项二
          </span>
        </div>
      </Checker>
    </Checkbox>
    <Checkbox
      bindGroup={true}
      key=".2"
      name="2"
    >
      <Checker
        bem={[Function]}
        bindGroup={true}
        checked={false}
        name="2"
        onToggle={[Function]}
        parent={
          Object {
            "props": Object {
              "children": Array [
                <Checkbox
                  bindGroup={true}
                  name="0"
                >
                  选项一
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  checked={true}
                  name="1"
                >
                  选项二
                </Checkbox>,
                <Checkbox
                  bindGroup={true}
                  name="2"
                >
                  选项三
                </Checkbox>,
              ],
              "defaultValue": Array [],
              "value": Array [
                "1",
              ],
            },
          }
        }
        role="checkbox"
        shape="round"
      >
        <div
          aria-checked={false}
          className="rc-checkbox"
          onClick={[Function]}
          role="checkbox"
          tabIndex={0}
        >
          <div
            className="rc-checkbox__icon rc-checkbox__icon--round"
            style={
              Object {
                "fontSize": undefined,
              }
            }
          >
            <Icon
              name="success"
              style={Object {}}
              tag="i"
            >
              <Badge
                className="van-icon van-icon-success"
                showZero={true}
                style={
                  Object {
                    "color": undefined,
                    "fontSize": undefined,
                  }
                }
                tag="i"
              >
                <i
                  className="rc-badge__wrapper van-icon van-icon-success"
                  style={
                    Object {
                      "color": undefined,
                      "fontSize": undefined,
                    }
                  }
                />
              </Badge>
            </Icon>
          </div>
          <span
            className="rc-checkbox__label"
          >
            选项三
          </span>
        </div>
      </Checker>
    </Checkbox>
  </div>
</ForwardRef>
`;
